src/components/elements/FlexibleNodeCard.tsx

```diff
@@ -0,0 +1,109 @@
+import { useRouter } from "next/router";
+import { Card, Row, Col, Typography } from "antd";
+import styles from "../../styles/Card.module.scss";
+import { Gateway, Node, Reading, SensorType } from "../../services/AppModel";
+import { NODE_MESSAGE } from "../../constants/ui";
+import { getFormattedLastSeenDate } from "../presentation/uiHelpers";
+import registry from "../renderers/client";
+import { ReadingVisualization } from "../renderers/renderers";
+import TextReadingRenderer from "./TextReadingRenderer";
+import TextReadingRendererComponent from "./TextReadingRenderer";
+
+interface NodeProps {
+  gateway: Gateway;
+  node: Node;
+  index: number;
+}
+
+const NodeCardComponent = (props: NodeProps) => {
+  const { gateway, node, index } = props;
+  const { Text } = Typography;
+ 
+  const router = useRouter();
+  // todo - use urlManager to construct the URL from the gateway/node. 
+  const nodeUrl = `/${gateway.id.gatewayDeviceUID}/node/${node.id.nodeID}/details`;
+  const handleCardClick = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    router.push(nodeUrl);
+  };
+
+  const name = node.name || NODE_MESSAGE.NO_NAME;
+
+  // todo - fix up location. Either as a property on Node or as a known sensor type
+  const location = NODE_MESSAGE.NO_LOCATION;
+  const lastActivity = node.lastSeen ? getFormattedLastSeenDate(new Date(node.lastSeen)) : NODE_MESSAGE.NEVER_SEEN;
+  const nodeReadings = node.currentReadings || [];
+
+  console.log("nodeReadings.length", nodeReadings.length, node.id.nodeID, nodeReadings);
```

Comment:
remove console.log?

Reply:
This is still WiP but when we get closer to the code being production ready we'll remove it. (Along with cleaning up other lint complaints!)